pyworkflow/object.py

```diff
@@ -1116,7 +1116,9 @@ def setMapperClass(self, MapperClass):
         
     def __getitem__(self, itemId):
         """ Get the image with the given id. """
-        return self._getMapper().selectById(itemId)
+        item = self._getMapper().selectById(itemId)
+        self.close()
```

Comment:
This can be a performance killer @pconesa. This is call everything time that you retrieve an item from a set. I think we should not close the mapper here but externally when the set is not longer used.

Reply:
I know, this is only happening when using the set as a set[23]

I couldn't find a better solution. At this moment is either this or nothing? And nothing would be having sqlite locks for, probably SetOfVolumes or any other set which items could be pointed from other protocols (I think atomStructfits here too).

Reply:
We can leave this here assuming the locks. I think they've been there for a long time. I can not relate this to other issues, like performance problems or errors.

If at some point we identify a relation between some issues and what this PR is trying to solve (#116 ), then, maybe we can rescue this or analyze the problem again?

@azazellochg , @delarosatrevin ?

Reply:
@pconesa Can you take a quick look where this set[id] is being used? If it is not in critical parts, maybe we can have the. mapper.close there...I was a bit confused with the iterItems, that is widely used and very very critical.

Reply:
I wish I knew how to take a quick look of usage of set[id] ....all I see is iteritems usage and get firstItem ....I don't think we are iterating a set using the index. But I might be wrong.

Reply:
I'm back with this. What about. I think the __item__ Method is rarely used in iterations. I've seen a couple of usages in gui code.

I'm also thinking about caching the item in a dict {id: item}, preventing the access that now, for example, is happening when filling the summary:
1 to create the I/O tree 
2 used in the methods section.